core/src/test/java/org/apache/iceberg/rest/requests/TestCreateNamespaceRequest.java

```diff
@@ -0,0 +1,128 @@
+import org.apache.iceberg.AssertHelpers;
+import org.apache.iceberg.catalog.Namespace;
+import org.apache.iceberg.exceptions.ValidationException;
+import org.apache.iceberg.relocated.com.google.common.collect.ImmutableMap;
+import org.apache.iceberg.rest.RESTUtils;
+import org.assertj.core.api.Assertions;
+import org.junit.Test;
+
+public class TestCreateNamespaceRequest {
+
+  private static final ObjectMapper MAPPER = RESTUtils.mapper();
+
+  private static final String[] ALL_FIELDS_FROM_SPEC = { "namespace", "properties" };
+
+  /* Values used to fill in request fields */
+  private static final Namespace NAMESPACE = Namespace.of("accounting", "tax");
+  private static final Map<String, String> PROPERTIES = ImmutableMap.of("owner", "Hank");
+  private static final Map<String, String> EMPTY_PROPERTIES = ImmutableMap.of();
+
+  @Test
+  public void testValidSerDe() throws JsonProcessingException {
+    Map<String, CreateNamespaceRequest> toTest = ImmutableMap.of(
+        // Full response
+        "{\"namespace\":[\"accounting\",\"tax\"],\"properties\":{\"owner\":\"Hank\"}}",
+        builder().withNamespace(NAMESPACE).setProperties(PROPERTIES).build(),
+        "{\"namespace\":[\"accounting\",\"tax\"],\"properties\":null}",
+        builder().withNamespace(NAMESPACE).build(),
+        "{\"namespace\":[\"accounting\",\"tax\"],\"properties\":{}}",
+        builder().withNamespace(NAMESPACE).setProperties(EMPTY_PROPERTIES).build());
+
+    for (Map.Entry<String, CreateNamespaceRequest> testCase : toTest.entrySet()) {
+      String json = testCase.getKey();
+      CreateNamespaceRequest req = testCase.getValue();
+      assertSerializes(req, json);
+      assertDeserializes(json, req);
+    }
+  }
+
+  @Test
+  public void testDeserializeInvalidRequest() {
+    String jsonIncorrectTypeForNamespace = "{\"namespace\":\"accounting%00tax\",\"properties\":null}";
+    AssertHelpers.assertThrows(
+        "A JSON request with incorrect types for fields should fail to deserialize and validate",
+        JsonProcessingException.class,
+        "Namespace must be an array",
+        () -> MAPPER.readValue(jsonIncorrectTypeForNamespace, CreateNamespaceRequest.class).validate()
+    );
+
+    String jsonIncorrectTypeForProperties = "{\"namespace\":[\"accounting\",\"tax\"],\"properties\":[]}";
+    AssertHelpers.assertThrows(
+        "A JSON request with incorrect types for fields should fail to parse and validate",
+        JsonProcessingException.class,
+        () -> MAPPER.readValue(jsonIncorrectTypeForProperties, CreateNamespaceRequest.class).validate()
+    );
+
+    String jsonMisspelledKeys = "{\"namepsace\":[\"accounting\",\"tax\"],\"propertiezzzz\":{\"owner\":\"Hank\"}}";
+    AssertHelpers.assertThrows(
+        "A JSON request with the keys spelled incorrectly should fail to deserialize and validate",
+        JsonProcessingException.class,
+        () -> MAPPER.readValue(jsonMisspelledKeys, CreateNamespaceRequest.class).validate()
```

Comment:
What fails about this? Is it that there are unknown keys in the request?

Reply:
I added an exception message to one of the invalid key tests.